pennylane/interfaces/execution.py

```diff
@@ -446,6 +446,8 @@ def cost_fn(params, x):
             if not tf.executing_eagerly() or "autograph" in interface:  # pragma: no cover
                 from .tensorflow_autograph import execute as _execute  # pragma: no cover
 
+                _execute = partial(_execute, mode=_mode)
+

```

Comment:
I added this back since some autograph tests were failing without passing in `mode`.